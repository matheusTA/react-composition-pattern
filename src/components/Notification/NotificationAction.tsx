import { ButtonHTMLAttributes, ElementType } from 'react'
import { twMerge } from 'tailwind-merge'

interface NotificationActionProps
  extends ButtonHTMLAttributes<HTMLButtonElement> {
  icon: ElementType
}

export function NotificationAction({
  icon: Icon,
  className,
  ...restProps
}: NotificationActionProps) {
  return (
    <button
      className={twMerge(
        'flex h-8 w-8 items-center justify-center rounded bg-zinc-400 hover:bg-zinc-500',
        className,
      )}
      {...restProps}
    >
      <Icon className="h-3 w-3 text-zinc-50" />
    </button>
  )
}
