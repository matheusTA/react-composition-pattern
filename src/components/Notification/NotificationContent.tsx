interface NotificationContentProps {
  text: string
}

export function NotificationContent({ text }: NotificationContentProps) {
  return (
    <div className="flex flex-1 flex-col gap-2">
      <p className="text-sm leading-relaxed text-zinc-600">{text}</p>

      <div className="flex items-center gap-1 text-xxs text-zinc-400">
        <span>Convite</span>
        <span>HÃ¡ 3 min</span>
      </div>
    </div>
  )
}
