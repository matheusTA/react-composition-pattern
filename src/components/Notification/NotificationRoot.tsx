import { ReactNode } from 'react'

interface NotificationRootProps {
  children: ReactNode
}

export function NotificationRoot({ children }: NotificationRootProps) {
  return (
    <div className="flex items-start gap-6 bg-zinc-200 px-8 py-4">
      {children}
    </div>
  )
}
