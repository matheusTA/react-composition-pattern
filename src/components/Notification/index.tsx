import { NotificationRoot } from './NotificationRoot'
import { NotificationIcon } from './NotificationIcon'
import { NotificationContent } from './NotificationContent'
import { NotificationActions } from './NotificationActions'
import { NotificationAction } from './NotificationAction'

export const Notification = {
  Root: NotificationRoot,
  Icon: NotificationIcon,
  Content: NotificationContent,
  Actions: NotificationActions,
  Action: NotificationAction,
}
