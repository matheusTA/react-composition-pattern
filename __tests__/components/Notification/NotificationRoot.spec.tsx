import { render } from '@testing-library/react'
import { NotificationRoot } from '@/components/Notification/NotificationRoot'

import '@testing-library/jest-dom'

describe('NotificationRoot', () => {
  it('renders NotificationRoot unchanged', () => {
    const children = (
      <>
        <div data-testid="child-1">Child</div>
      </>
    )

    const { container } = render(
      <NotificationRoot>{children}</NotificationRoot>,
    )

    expect(container).toMatchSnapshot()
  })

  it('renders children correctly', () => {
    const children = (
      <>
        <div data-testid="child-1">Child</div>
      </>
    )

    const { getByText } = render(
      <NotificationRoot>{children}</NotificationRoot>,
    )

    const childElement = getByText('Child')

    expect(childElement).toBeInTheDocument()
  })

  it('applies the correct CSS class to the container div', () => {
    const children = (
      <>
        <div>Child 1</div>
        <div>Child 2</div>
      </>
    )

    const { container } = render(
      <NotificationRoot>{children}</NotificationRoot>,
    )

    const notificationRootDiv = container.firstChild

    expect(notificationRootDiv).toBeInTheDocument()
    expect(notificationRootDiv).toHaveClass(
      'flex items-start gap-6 bg-zinc-200 px-8 py-4',
    )
  })
})
