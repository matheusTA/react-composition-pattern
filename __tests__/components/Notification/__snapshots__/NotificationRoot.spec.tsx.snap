// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationRoot renders NotificationRoot unchanged 1`] = `
<div>
  <div
    class="flex items-start gap-6 bg-zinc-200 px-8 py-4"
  >
    <div
      data-testid="child-1"
    >
      Child
    </div>
  </div>
</div>
`;
